tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/bluetooth.py

```diff
@@ -34,16 +33,18 @@ def device_name(self):
 
     def enable_bluetooth(self):
         self.marionette.find_element(*self._bluetooth_label_locator).tap()
-        checkbox = self.marionette.find_element(*self._bluetooth_checkbox_locator)
-        Wait(self.marionette).until(expected.element_selected(checkbox))
         rename_device = self.marionette.find_element(*self._rename_my_device_button_locator)
         Wait(self.marionette).until(expected.element_enabled(rename_device))
 
-    def enable_visible_to_all(self):
-        # Bluetooth state is stored outside the profile bug 969310
+    def disable_bluetooth(self):
+        rename_device = self.marionette.find_element(*self._rename_my_device_button_locator)
+        self.marionette.find_element(*self._bluetooth_label_locator).tap()
+        Wait(self.marionette).until(
+            expected.element_not_displayed(rename_device))
+
+    def enable_visible_to_all(self, data_layer):
         self.marionette.find_element(*self._visible_to_all_label_locator).tap()
-        checkbox = self.marionette.find_element(*self._visible_to_all_checkbox_locator)
-        Wait(self.marionette).until(expected.element_selected(checkbox))
+        Wait(self.marionette).until(lambda m: data_layer.bluetooth_is_discoverable is True)
```

Comment:
same as above, it's either I don't know how to check the checkbox, or the page doesn't use the checkbox in a conventional manner